components/raftstore/src/store/fsm/peer.rs

```diff
@@ -3012,6 +3012,31 @@ where
             // It's v2 only message and ignore does no harm.
             ExtraMessageType::MsgGcPeerResponse | ExtraMessageType::MsgFlushMemtable => (),
             ExtraMessageType::MsgRefreshBuckets => self.on_msg_refresh_buckets(msg),
+            ExtraMessageType::MsgSnapGenPrecheckRequest => {
+                let result = self.ctx.snap_mgr.recv_snap_precheck(msg.region_id);
+                self.fsm.peer.send_snap_gen_precheck_response(
+                    self.ctx,
+                    &msg.from_peer.unwrap(),
+                    result,
+                )
+            }
+            ExtraMessageType::MsgSnapGenPrecheckResponse => {
+                let result = msg.get_extra_msg().get_snap_gen_precheck_passed();
+                fail_point!("snap_gen_precheck_failed", !result, |_| {});
+                info!(
+                    "snap gen precheck response: {}", result;
+                    "region_id" => self.region_id(),
+                    "peer_id" => self.peer().id,
+                    "store_id" => self.store_id(),
+                    "receiver_peer_id" => msg.get_from_peer().get_id(),
+                    "receiver_store_id" => msg.get_from_peer().get_store_id(),
+                );
+                if result {
+                    if let Some(gen_task) = self.fsm.peer.mut_store().mut_gen_snap_task() {
+                        gen_task.precheck_succeeded = true
```

Comment:
Can we schedule apply gen task here?

Reply:
good point. that way we don't need the `precheck_succeeded` field. made that change in the latest commit